plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -89,7 +89,42 @@ func LeastRequestedPriority(pod *api.Pod, podLister algorithm.PodLister, minionL
 
 	list := algorithm.HostPriorityList{}
 	for _, node := range nodes.Items {
-		list = append(list, calculateOccupancy(pod, node, podsToMachines[node.Name]))
+		list = append(list, calculateResourceOccupancy(pod, node, podsToMachines[node.Name]))
+	}
+	return list, nil
+}
+
+func min(l, r int64) (m int64) {
+	m = r
+	if l < r {
+		m = l
+	}
+	return m
+}
+
+// See comment for DumbSpreadingPriority()
+const dumbSpreadingDenominator int64 = 10
+
+// DumbSpreadingPriority is a priority function that favors nodes with fewer pods.
+// It works like LeastRequestedPeriority but instead of using 10 * percentage of machine free by resource,
+// it uses 10 * percentage of machine free by pod, with "percentage of machine free by pod" claculated as
+// (dumbSpreadingDenominator - number of pods already on the node + 1) / dumbSpreadingDenominator.
+// dumbSpreadingDenominator serves like the machine capacity in LeasRequestedPriority but is chosen
+// so that we equate one pod with a reasonable amount of resources when we combine all the scores together.
```

Comment:
@bprashanth I re-read your initial comment and now I understand what you were saying -- instead of adding a spreading algorithm, just treat limit-0 pods as if they had some non-zero limit (maybe 5% or 10% of the capacity of the machine or something). I remember we did talk about that on the issue, but I think it could still have drawbacks (for example if the limit difference between two machines is 50%, it will put 5 or 10 pods on one machine before moving to the other machine).

An alternative would be something you suggested earlier: to use dumb spreading for limit-0 pods and the resource-based priority algorithms for the pods with limits. I'm starting to think that might be the way to go.